### PR TITLE
Refactoring of navigation layout and patching several issues

### LINKED ISSUE #50: I would like to be able to post favors on the same screen as in the map tab.
floating button

### LINKED ISSUE #147: The offline button shown on the map overlaps with the "current position" button

### LINKED ISSUE #160: The bottom bar in the main page is not a real bottom navigation menu and this affect the UI in all the pages related to the navigation drawer (e.g. changing the color of the bar is reflected in pages where the tab bar should not be visible))

### LINKED ISSUE #161: As a user, I would like to open the hamburger menu from any page of the navigation drawer and not be forced to go back to the home page to re-open the menu.

### LINKED ISSUE #162: Pressing the Android back button does not make you go back to the home page but to the previous page visited. As a user, I would like to always go back to the home page from any page of the navigation drawer.

### PR DESCRIPTION
**Pull request by mikanikos**
The PR refactors the current implementation of the tab bar with a real bottom navigation menu and make several changes to the navigation layout: a toolbar is now used and is always invisible (and the background clickable) and allows us to use the native app bar items without relying on custom clickable images (back arrow, hamburger button etc.). With this addition, fragments in the navigation drawer don't need to implement any interface or methods and they can just be added in the menu from the MainActivity. The result is not strictly visible because I tried to leave things as much as before but the UI looks much more polished thanks to the native app bar items and the real bottom navigation menu. However, the main advantage is that the implementation is much simpler now and it will be easier for us to work on it in the future.

The favors list page is refactored with the addition of menu items in the invisible app bar in order to better organize the UI. The search view widget library is replaced with the newer androidx library which provides a better handling of the exit from the search view and prepare the next PR based on the synchronization of the list with the database (unique downside is that the state of the search view is not saved when clicking on an item inside the search view but this will be better incorporated in the sync list PR).

Other smaller notes: the image in the navigation drawer is changed with the app logo; the floating button to add a new favor is added to the map page as well, the offline map button is moved to the bottom left corner because it was overlapping with the current position button. 

Closes #160, closes #161, closes #162, closes #147, closes #50.

The PR includes a patch that refactors constraints and margins between items (https://github.com/favo-sdp/Favo/pull/159).

### CODE REVIEW

File: app/src/main/java/ch/epfl/favo/view/tabs/FavorPage.java [hunk 1]
```diff
@@ -168,16 +109,112 @@ private void setupListView() {
...
+
+    // Inflate the menu; this adds items to the action bar if it is present.
+    inflater.inflate(R.menu.options_menu, menu);
+    super.onCreateOptionsMenu(menu, inflater);
+
+    setupSpinner(menu);
+    setupSearch(menu);
+  }
+
+  private void setupSearch(Menu menu) {
```
[the remark this answers is not in the document]
Comment (mikanikos) [reply app/src/main/java/ch/epfl/favo/view/tabs/FavorPage.java:153]: Solved

File: app/src/main/java/ch/epfl/favo/view/tabs/FavorPage.java [hunk 2]
```diff
@@ -168,16 +109,112 @@ private void setupListView() {
...
+          public boolean onQueryTextChange(String newText) {
+            // replace irrelevant items on listView with last query results or empty view
+
+            // displayFavorList(new HashMap<>(), R.string.empty);
+            return false;
+          }
+        });
+  }
+
+  private void setupSpinner(Menu menu) {
```
[the remark this answers is not in the document]
Comment (mikanikos) [reply app/src/main/java/ch/epfl/favo/view/tabs/FavorPage.java:211]: Solved

File: app/src/main/res/layout/mainactivity/layout/content_main.xml [hunk 3]
```diff
@@ -1,26 +1,67 @@
...
-    <include
-        android:id="@+id/bot_view"
-        layout="@layout/bottom_navigation_view"
-        android:layout_width="match_parent"
-        android:layout_height="wrap_content"
-        android:layout_alignParentBottom="true" />
-
-</RelativeLayout>
\ No newline at end of file
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (danielpgp1012) [inline app/src/main/res/layout/mainactivity/layout/content_main.xml:1]: Nice this layout is indeed cleaner

**Comment by mikanikos** [comment]
@huajian1069 
> 
> navigation from favor list to map is changed to use popbackstack(); 
You're right, that's a good idea!

> bottomNavigation is hidden in FavorDetail view;
I actually hid the bottomNavigation purposely, do we really want to have it in the favorDetail and favroRequest pages? @danielpgp1012 @muneebazam @RootofalleviI @arosu  

> display correct active/archived list after quitting from search mode.
Don't worry about this, I actually left it like this because I completely refactored that part in my next PR where I sync the list with the db and implement full-text search.

**Pull request merged**